src/aco.cpp

```diff
@@ -196,10 +206,50 @@ namespace full_coverage_path_planner
                         }
                     }
 
+                    //move ant
                     visited[new_location.pos.y][new_location.pos.x] = eNodeVisited;
                     ant.move(new_location, true);
                     ant.visited_counter++;
                     goals = map_2_goals(visited, eNodeOpen);
+
+                    //move ant extra according to velocity
+                    int moved = 1;
+                    int nx = new_location.pos.x;
+                    int ny = new_location.pos.y;
+
+                    while (moved != ant.velocity)
+                    {
+                        if (ant.direction == NORTH)
+                        {
+                            ny++;
+                        }
+                        if (ant.direction == EAST)
+                        {
+                            nx++;
+                        }
+                        if (ant.direction == SOUTH)
+                        {
+                            ny--;
+                        }
+                        if (ant.direction == WEST)
+                        {
+                            nx--;
+                        }
+
```

Comment:
If the ant moved North, move it North again.

If it moved South, move it South again.

etc.